File: 0226-invert-binary-tree/0226-invert-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        recursive(root);
        return root;
    }
    
    private void recursive(TreeNode node) {
        if (node == null) {
            return;
        }
        
        TreeNode temp = node.left;
        node.left = node.right;
        node.right = temp;
        recursive(node.left);
        recursive(node.right);
    }
}

File: 0226-invert-binary-tree/0226-invert-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
    return swap(root)
}

func swap(node *TreeNode) *TreeNode {
    if node == nil {
        return node
    }
    temp := node.Left
    node.Left = node.Right;
    node.Right = temp;
    
    swap(node.Left)
    swap(node.Right)
    return node
}